pkg/util/provider/machinecontroller/machineclass.go

```diff
@@ -146,24 +146,34 @@ func (c *controller) reconcileClusterMachineClass(class *v1alpha1.MachineClass)
 		return err
 	}
 
-	// fetch all machines referring the machineClass
+	// Fetch all machines referring the machineClass
 	machines, err := c.findMachinesForClass(machineutils.MachineClassKind, class.Name)
 	if err != nil {
 		return err
 	}
 
-	// Add finalizer to avoid losing machineClass object
 	if class.DeletionTimestamp == nil && len(machines) > 0 {
-		err = c.addMachineClassFinalizers(class)
-		if err != nil {
-			return err
+		// If deletionTimestamp is not set and more than one machines are referring this machineClass
```

Comment:
The comment should have been `// If deletionTimestamp is not set and at least one machine is referring this machineClass`

Reply:
Done - https://github.com/gardener/machine-controller-manager/pull/626/commits/8051ea143d76fc05bef4d5acd775fa8a3e5b8b8d#diff-65af0339f308cb1ad849a3a1f0d811395b2469d403084c0a2f1ba10f040ff63fR156.